typings/index.d.ts

```diff
@@ -1341,13 +1345,13 @@ declare module 'discord.js' {
     public send(message: any): Promise<Shard>;
     public spawn(spawnTimeout?: number): Promise<ChildProcess>;
 
-    public on(event: 'spawn' | 'death', listener: (child: ChildProcess) => void): this;
+    public on(event: 'spawn' | 'death' | 'parentDeath', listener: (child: ChildProcess) => void): this;
```

Comment:
all these events emit a ChildProcess OR a worker thread.. can you update typings to make them accurate?

Reply:
also enforced types on spawn and respawn, 👍🏻